### PR TITLE
disable fallback annotation logic when the controller is healthy

### LINKED ISSUE #826: Containers restart after Kuberarmor install
## Bug Report

**General Information**

As far as I know, prior to version 0.5, deployments were edited manually to apply the required annotation for KubeArmor. But this method is not applicable, for example, when using a database inside a container. The data was frayed, as a result, the database was corrupted. After version 0.5, annotations were added using the admission controller, which in fact should not lead to restarting the pod and corrupting the database in the future. Until some time, everything worked for me as it should, but after updating 0.5.5, the containers began to restart again, which entailed the destruction of the database.

### LINKED ISSUE #840: Disable kubearmor annotation fallback logic when annotation controller is active.
## Feature Request

**Short Description**

We want to prevent kubearmor fallback logic from executing when the annotation controller is present.

**Describe the solution you'd like**

~~The annotation controller will add to any new deployment the label `kubearmor-policy: enabled` and add a check in the fallback logic to not run if that annotation exist at the deployment level.~~

KubeArmor fallback logic will run only if the pod does not have all the required annotations. (one annotation per container).

### PR DESCRIPTION
**Pull request by achrefbensaad**
fixes #826, fixes #840 

We want to disable kubearmor annotation patch logic (fallback logic) when the annotations controller is working as expected.
Currently the fallback logic causes deployments restarts regardless of whether the annotation controller is present or not.

Signed-off-by: achref ben saad <achref9612@gmail.com>

### CODE REVIEW

**Comment by daemon1024** [comment]
Can you briefly explain what the code change is doing. I don't really understand anything seeing the changes 😅

File: KubeArmor/core/kubeUpdate.go [hunk 1]
```diff
@@ -606,14 +609,15 @@ func (dm *KubeArmorDaemon) WatchK8sPods() {
...
+						containersCount += 1
 					}
 
 					if event.Type == "ADDED" {
 						// update apparmor profiles
 						dm.RuntimeEnforcer.UpdateAppArmorProfiles(pod.Metadata["podName"], "ADDED", appArmorAnnotations)
 
 						if updateAppArmor && pod.Annotations["kubearmor-policy"] == "enabled" {
-							if deploymentName, ok := pod.Metadata["deploymentName"]; ok {
+							if deploymentName, ok := pod.Metadata["deploymentName"]; ok && containersCount != appArmorAnnotationsCount {
```
Comment (daemon1024) [inline KubeArmor/core/kubeUpdate.go:620]: We only patch deployment if annotation don't exist already?

Comment (achrefbensaad) [reply KubeArmor/core/kubeUpdate.go:620]: yes. If the annotation does not exist or it is uncomplete then we need to patch the deployment as reaching this condition means that the annotation controller is not healthy.

**Comment by achrefbensaad** [comment]
> Can you briefly explain what the code change is doing. I don't really understand anything seeing the changes 😅

@daemon1024, I have updated the PR description.

**Review by Ankurk99 — APPROVED** [review]
LGTM!

**Review by daemon1024 — APPROVED** [review]

**Pull request merged**